### PR TITLE
feat: diagnostics_channel channel.publish()

### PR DESCRIPTION
**Pull request by davecardwell**
Adds the missing defintion for the `Channel.prototype.publish` function.

https://nodejs.org/docs/latest-v16.x/api/diagnostics_channel.html#channelpublishmessage

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/docs/latest-v16.x/api/diagnostics_channel.html#channelpublishmessage
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by davecardwell** [comment]
The CI failures are coming from @types/mongodb, unrelated to this PR. What’s the process for getting past this?

(cc/ @peterblazejewicz)

**Comment by peterblazejewicz** [comment]
> The CI failures are coming from @types/mongodb, unrelated to this PR. What’s the process for getting past this?

this is not directly related to DT project, those types come from 3rd party libraries that are using @types/* and are referenced in DT projects. MS team folks are directly working on fixing those.

TL;DR;  Definitely Typed is a proofing ground for next iteration of TS to catch those kind of errors early. The most likely project to be impacted, is @types/node. Will be fixed automatically by maintainer

File: types/node/index.d.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@
 //                 NodeJS Contributors <https://github.com/NodeJS>
 //                 Linus Unnebäck <https://github.com/LinusU>
 //                 wafuwafu13 <https://github.com/wafuwafu13>
+//                 Dave Cardwell <https://github.com/davecardwell>
```
Comment (peterblazejewicz) [inline types/node/index.d.ts:43]: fact check: you will be notified on any future updates to a code as part of code reviews and your input on changes will be welcome (a.k.a. you'll become owner of the implementation here)
is that your intent?

Comment (davecardwell) [reply types/node/index.d.ts:43]: @peterblazejewicz Yes that’s no problem, I have a this on a few other type definitions already so am familiar with the process.

**Comment by peterblazejewicz** [comment]
@davecardwell 
I think there is a problem with current subscription details here for listener (`onMessage`), it's wrongly typed in existing code. I'll create PR to fix it and it should not impact your change. I'll post a cross-ref

**Comment by peterblazejewicz** [comment]
x-ref: #59796

**Review by peterblazejewicz — APPROVED** [review]

**Comment by davecardwell** [comment]
Ready to merge

Thanks, @peterblazejewicz!
